lib/Service/RecentPagesService.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Collectives\Service;
+
+use OCA\Collectives\Model\PageInfo;
+use OCA\Collectives\Model\RecentPage;
+use OCP\DB\Exception;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\Files\IMimeTypeLoader;
+use OCP\IConfig;
+use OCP\IDBConnection;
+use OCP\IURLGenerator;
+use OCP\IUser;
+
+class RecentPagesService {
+
+	public function __construct(
+		protected CollectiveService $collectiveService,
```

Comment:
Given that we still target PHP 7.4, constructor property promotion is not possible in Collectives yet.